### PR TITLE
Get rid of some remaining instances of C numeric limits

### PR DESCRIPTION
**Pull request by Capostrophic**
Yep, why not. I didn't touch recastnavigation and defines in sound system.

Also fixed an intentional unreasonable overflow in land vertex color default values.

### CODE REVIEW

**Comment by akortunov** [comment]
Looks good to me. Let see if CI will build it fine...

File: components/esm/loadland.cpp [hunk 1]
```diff
@@ -200,9 +201,9 @@ namespace ESM
             mLandData->mTextures[i] = 0;
         for (int i = 0; i < LAND_NUM_VERTS; ++i)
         {
-            mLandData->mColours[i*3+0] = -1;
-            mLandData->mColours[i*3+1] = -1;
-            mLandData->mColours[i*3+2] = -1;
```
Comment (akortunov) [inline components/esm/loadland.cpp]: I wonder why compilers did not complain - mColours is an array of unsigned chars.

**Comment by psi29a** [comment]
I like as well, waiting for CI.

File: apps/opencs/view/world/util.cpp [hunk 2]
```diff
@@ -234,7 +235,7 @@ QWidget *CSVWorld::CommandDelegate::createEditor (QWidget *parent, const QStyleO
         case CSMWorld::ColumnBase::Display_Double:
         {
             DialogueDoubleSpinBox *dsb = new DialogueDoubleSpinBox(parent);
-            dsb->setRange(-FLT_MAX, FLT_MAX);
+            dsb->setRange(std::numeric_limits<double>::min(), std::numeric_limits<double>::max());
```
Comment (terabyte25) [inline apps/opencs/view/world/util.cpp]: Same as above from akortunov.

File: components/esm/loadland.cpp [hunk 3]
```diff
@@ -252,8 +253,8 @@ namespace ESM
         if (reader.isNextSub("VHGT")) {
             VHGT vhgt;
             if (condLoad(reader, flags, target->mDataLoaded, DATA_VHGT, &vhgt, sizeof(vhgt))) {
-                target->mMinHeight = FLT_MAX;
-                target->mMaxHeight = -FLT_MAX;
+                target->mMinHeight = std::numeric_limits<float>::max();
+                target->mMaxHeight = std::numeric_limits<float>::min();
```
Comment (terabyte25) [inline components/esm/loadland.cpp]: This as well.

**Pull request merged**